projects/extension/sql/idempotent/012-vectorizer-int.sql

```diff
@@ -465,76 +459,155 @@ $func$
+    into strict _pk_change_check
+    from pg_catalog.jsonb_to_recordset(source_pk) x(attnum int, attname name);
+    _func_def := $def$
+    begin
+        if (TG_LEVEL = 'ROW') then
+            if (TG_OP = 'DELETE') then
+                $DELETE_STATEMENT$;
+            elsif (TG_OP = 'UPDATE') then
+                if $PK_CHANGE_CHECK$ then
+                    $DELETE_STATEMENT$;
+                end if;
+                
+                insert into $QUEUE_SCHEMA$.$QUEUE_TABLE$ ($PK_COLUMNS$)
+                values ($PK_VALUES$);
+                return new;
+            else
+                insert into $QUEUE_SCHEMA$.$QUEUE_TABLE$ ($PK_COLUMNS$)
+                values ($PK_VALUES$);
+                return new;
+            end if;
+
+        elsif (TG_LEVEL = 'STATEMENT') then
+            if (TG_OP = 'TRUNCATE') then
+                execute format('truncate table %I.%I', '$TARGET_SCHEMA$', '$TARGET_TABLE$');
+                execute format('truncate table %I.%I', '$QUEUE_SCHEMA$', '$QUEUE_TABLE$');
+            end if;
+            return null;
+        end if;
+        
+        return null;
+    end;
+    $def$;
+
+    -- Replace placeholders
+    _func_def := replace(_func_def, '$DELETE_STATEMENT$', _delete_statement);
+    _func_def := replace(_func_def, '$PK_CHANGE_CHECK$', _pk_change_check);
+    _func_def := replace(_func_def, '$QUEUE_SCHEMA$', quote_ident(queue_schema));
+    _func_def := replace(_func_def, '$QUEUE_TABLE$', quote_ident(queue_table));
+    _func_def := replace(_func_def, '$PK_COLUMNS$', _pk_columns);
+    _func_def := replace(_func_def, '$PK_VALUES$', _pk_values);
+    _func_def := replace(_func_def, '$TARGET_SCHEMA$', quote_ident(target_schema));
+    _func_def := replace(_func_def, '$TARGET_TABLE$', quote_ident(target_table));
+
+    return _func_def;
+end;
+$func$ language plpgsql immutable security invoker
+set search_path to pg_catalog, pg_temp;
 
 -------------------------------------------------------------------------------
 -- _vectorizer_create_source_trigger
 create or replace function ai._vectorizer_create_source_trigger
-( trigger_name pg_catalog.name
-, queue_schema pg_catalog.name
-, queue_table pg_catalog.name
-, source_schema pg_catalog.name
-, source_table pg_catalog.name
-, source_pk pg_catalog.jsonb
+( trigger_name pg_catalog.name     -- Name for the trigger
+, queue_schema pg_catalog.name     -- Schema containing the queue table
+, queue_table pg_catalog.name      -- Table that will store queued items
+, source_schema pg_catalog.name    -- Schema containing the watched table
+, source_table pg_catalog.name     -- Table being watched for changes
+, target_schema pg_catalog.name    -- Schema containing the target table for deletions
+, target_table pg_catalog.name     -- Table where corresponding rows should be deleted
+, source_pk pg_catalog.jsonb       -- JSON describing primary key columns to track
+, original_owner pg_catalog.name default null   -- Role that should own the trigger
```

Comment:
`original_owner` is unused

Reply:
do you have some sort of plugin for pycharm that helps with this kinda stuff? I am not used to scanning for this myself 😅

Reply:
The text shows up in a lighter color when it's unused.